recipes/freexl/all/conanfile.py

```diff
@@ -36,20 +38,30 @@ def requirements(self):
         self.requires("libiconv/1.16")
 
     def build_requirements(self):
-        if tools.os_info.is_windows and self.settings.compiler != "Visual Studio" and \
-           "CONAN_BASH_PATH" not in os.environ and tools.os_info.detect_windows_subsystem() != "msys2":
-            self.build_requires("msys2/20200517")
+        if self.settings.compiler != "Visual Studio":
+            self.build_requires("gnu-config/cci.20201022")
+            if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+                self.build_requires("msys2/cci.latest")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename(self.name + "-" + self.version, self._source_subfolder)
 
+    @property 
+    def _user_info_build(self): 
+        return getattr(self, "user_info_build", None) or self.deps_user_info 
+
     def build(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
+
         if self.settings.compiler == "Visual Studio":
             self._build_msvc()
         else:
+            shutil.copy(self._user_info_build["gnu-config"].CONFIG_SUB,
+                    os.path.join(self._source_subfolder, "config.sub"))
+            shutil.copy(self._user_info_build["gnu-config"].CONFIG_GUESS,
+                    os.path.join(self._source_subfolder, "config.guess"))
```

Comment:
It would be nice to let (build) requirements make available arbitrary python functions/objects.
Right now, only strings can be passed.
This would allow the following usage:
```python
self.deps_user_info["gnu-config"].copy_config_files(self._source_folder)
```